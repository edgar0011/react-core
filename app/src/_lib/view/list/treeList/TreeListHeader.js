import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

// TODO: move to core-fe
/**
 * @class
 * @name TreeListHeader
 * @desc Tree List Header component.
 * @prop {Object[]} columns Required columns to display.
 * @prop {string} title Optional header title string.
 * @prop {object} nodeStyle Optional style of header node.
 * @prop {string} justify Optional flex justify representation.
 * @prop {function} render Render function must be specified unless RenderComponent is specified.
 * @prop {ComponentType} RenderComponent Render component must be specified unless render function is specified.
 */
export default class TreeListHeader extends PureComponent {
  static propTypes = {
    columns: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
      label: PropTypes.string
    })).isRequired,
    title: PropTypes.string,
    nodeStyle: PropTypes.object,
    justify: PropTypes.string,
    render: PropTypes.func,
    RenderComponent: PropTypes.any
  }

  static defaultProps = {
    title: ''
  }

  render() {
    const { render, RenderComponent, ...props } = this.props

    let rendered
    if (render) {
      rendered = render({ ...props })
    } else {
      rendered = (
        <RenderComponent
          {...props}
        />
      )
    }
    return rendered
  }
}
