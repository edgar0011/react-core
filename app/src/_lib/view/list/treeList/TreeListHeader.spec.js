// TODO: move to core-fe
import React from 'react'
import TreeListHeader from './TreeListHeader'

describe('TreeListHeader', () => {
  it('should render TreeListHeader', () => {
    global.shallowSnapshot(<TreeListHeader />)
  })
})
