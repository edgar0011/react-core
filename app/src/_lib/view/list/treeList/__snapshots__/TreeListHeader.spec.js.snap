// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeListHeader should render TreeListHeader 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TreeListHeader
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "title": "",
    },
    "ref": null,
    "rendered": null,
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "title": "",
      },
      "ref": null,
      "rendered": null,
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
